pageserver/src/tenant.rs

```diff
@@ -3405,30 +3410,42 @@ fn run_initdb(
         initdb_bin_path, initdb_target_dir, initdb_lib_dir,
     );
 
-    let initdb_output = Command::new(&initdb_bin_path)
+    let _permit = INIT_DB_SEMAPHORE.acquire().await;
+
+    let initdb_command = tokio::process::Command::new(&initdb_bin_path)
         .args(["-D", initdb_target_dir.as_ref()])
         .args(["-U", &conf.superuser])
         .args(["-E", "utf8"])
         .arg("--no-instructions")
-        // This is only used for a temporary installation that is deleted shortly after,
-        // so no need to fsync it
         .arg("--no-sync")
         .env_clear()
         .env("LD_LIBRARY_PATH", &initdb_lib_dir)
         .env("DYLD_LIBRARY_PATH", &initdb_lib_dir)
         .stdout(Stdio::null())
-        .output()
+        .spawn()
```

Comment:
According to docs, using `spawn()` here without redirecting stderr means the stderr output of initdb will spew into the stderr file descriptor of the parent, i.e., the pageserver process.

This means we won't see the output in the `initdb failed` bail below.

Reply:
redirected to Stdio::null stderr as well

Reply:
Note that if you "redirect it to null", on error we would see nothing on the `initdb failed`. It needs to be `Stdio::piped()`, so that `wait_with_output` will gather it as a Vec which we can later format as a string.